### PR TITLE
PM-12779: Add account security set up unlock method action card

### PR DESCRIPTION
**Pull request by matt-livefront**
## 🎟️ Tracking

[PM-12779](https://bitwarden.atlassian.net/browse/PM-12779)

## 📔 Objective

Adds a new action card to the account security settings menu for setting up an unlock method.

## 📸 Screenshots

https://github.com/user-attachments/assets/c267be6c-ff9b-429a-a2b3-fb703db5aeb4

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-12779]: https://bitwarden.atlassian.net/browse/PM-12779?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Platform/Application/Appearance/StyleGuideFont.swift [hunk 1]
```diff
@@ -112,28 +112,49 @@ extension View {
     ///
     /// - Parameters:
     ///   - style: The style of the text.
+    ///   - includeLinePadding: A flag to indicate if the style should apply padding around the
+    ///     view to account for the font's line height. Defaults to `true`.
```
Comment (matt-livefront) [inline BitwardenShared/UI/Platform/Application/Appearance/StyleGuideFont.swift:116]: The style guide modifier adds extra padding to text to try and match the line height of the font in Figma. But this was causing some issues when trying to align the views at the top of the action card, where the text was too low compared to the badge and close button.

Simulator Screenshot - iPhone 15 Pro - 2024-09-27 at 16 47 00

File: BitwardenShared/UI/Platform/Application/Views/BitwardenBadge.swift [hunk 2]
```diff
@@ -12,18 +12,34 @@ struct BitwardenBadge: View {
...
+            .styleGuide(.callout, weight: .bold, includeLinePadding: false, includeLineSpacing: false)
             .foregroundStyle(Asset.Colors.iconBadgeForeground.swiftUIColor)
-            .padding(badgePadding)
+            .onSizeChanged { textSize = $0 }
+            // Add custom horizontal and vertical padding to ensure the frame of the badge is a
+            // square. When the view is clipped to a circle, this ensures the frame of the view
+            // matches the size of the circle. Otherwise, clipping can cause there to be extra
+            // space in the frame which can affect the layout when the badge is used.
+            .padding(.horizontal, (circleDiameter - textSize.width) / 2 + badgePadding)
+            .padding(.vertical, (circleDiameter - textSize.height) / 2 + badgePadding)
```
Comment (matt-livefront) [inline BitwardenShared/UI/Platform/Application/Views/BitwardenBadge.swift:42]: Without these changes, the frame of the badge had extra vertical padding when displaying text that was taller than wider. This was also causing alignment issues at the top of action card.

| Before | After |
| --- | --- |
| Screenshot 2024-09-27 at 4 41 10 PM | Screenshot 2024-09-27 at 4 43 19 PM |

**Review by fedemkr — DISMISSED** [review]
⛏️ There seem to be some Codecov tests missing, I'll defer to you as to whether include them or not given that some of them seem very trivial.

**Comment by matt-livefront** [comment]
> ⛏️ There seem to be some Codecov tests missing, I'll defer to you as to whether include them or not given that some of them seem very trivial.

Thanks, I added tests for the action card. The view extension would be hard to test.

**Review by fedemkr — APPROVED** [review]

**Pull request merged**